Entities/MOSRotating.h

```diff
@@ -517,71 +569,97 @@ ClassInfoGetters
+    /// <param name="includeAttachablesWithNoDamageMultiplier">Whether to count wounds from Attachables that a zero damage multiplier, i.e. those that do not affect their parent (this MOSRotating) when wounded.</param>
+    /// <returns>The wound limit of this MOSRotating and, optionally, its Attachables.</returns>
+    int GetGibWoundLimit(bool includeAttachablesWithAPositiveDamageMultiplier, bool includeAttachablesWithANegativeDamageMultiplier, bool includeAttachablesWithNoDamageMultiplier) const;
 
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:  SetGibWoundLimit
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Set new wound limit, current wounds are not affected.
-// Arguments:       New wound limit.
-// Return value:    None.
+    /// <summary>
+    /// Sets the gib wound limit for this MOSRotating, i.e. the total number of wounds required to gib this MOSRotating.
+    /// This will not directly trigger gibbing, even if the limit is lower than the current number of wounds.
+    /// </summary>
+    /// <param name="newLimit">The new gib wound limit to use.</param>
+    void SetGibWoundLimit(int newGibWoundLimit) { m_GibWoundLimit = newGibWoundLimit; }
 
-	void SetGibWoundLimit(int newLimit) { m_GibWoundLimit = newLimit; }
+    /// <summary>
+    /// Gets the gib blast strength this MOSRotating, i.e. the strength with which Gibs and Attachables will be launched when this MOSRotating is gibbed.
+    /// </summary>
+    /// <returns>The gib blast strength of this MOSRotating.</returns>
+	int GetGibBlastStrength() const { return m_GibBlastStrength; }
 
+    /// <summary>
+    /// Sets the gib blast strength this MOSRotating, i.e. the strength with which Gibs and Attachables will be launched when this MOSRotating is gibbed.
+    /// </summary>
+    /// <param name="newGibBlastStrength">The new gib blast strength to use.</param>
+    void SetGibBlastStrength(int newGibBlastStrength) { m_GibBlastStrength = newGibBlastStrength; }
 
-	/// <summary>
-	/// Attaches the passed in wound AEmitter and adds it to the list of wounds, changing its parent offset to the passed in Vector.
-	/// </summary>
-	/// <param name="pWound">The wound AEmitter to add</param>
-	/// <param name="parentOffsetToSet">The vector to set as the wound AEmitter's parent offset</param>
-	void AddWound(AEmitter *pWound, const Vector& parentOffsetToSet, bool checkGibWoundLimit = true);
+    /// <summary>
+    /// Gets the number of wounds attached to this MOSRotating.
+    /// Includes any Attachables (and their Attachables, etc.) that have a positive damage multiplier.
+    /// <returns>The number of wounds on this MOSRotating.</returns>
+    /// </summary>
+    int GetWoundCount() const { return GetWoundCount(true, false, false); }
```

Comment:
I have mixed feelings about all of this.
Counting based on damage multipliers just seems confusing, plus the naming.
If it's not returning impulse/wound count/limit/whatever for this object and this object only it should be renamed to `GetTotalWhatever`.
I can't even properly figure all this shit out.
From the looks of it i'm never able to get a value for *this* object specifically without passing in arguments, this should be the other way around.
If you don't want to split this up or do any renaming then this definitely should default false everything so it returns wound count for this object only.

And this especially applies to `GetGibWoundLimit` which is a bigger culprit, and actually does need to be split up into a `GetGibWoundLimit` and `GetTotalGibWoundLimit`.
Printing `self.GibWoundLimit` on a brainbot torso spits out 135 instead of the ini defined 20.
Printing `self:GetGibWoundLimit(false, false, false)` spits out 93 and i'm honestly not even sure where that number comes from.
Printing `self.GibWoundLimit` after `self.GibWoundLimit = 10` on the torso spits out 52 I don't even. :judge: :judge: :judge: 
These maffs don't seem right to me, and they're probably the same for `WoundCount` and `GetWoundCount`.

Also having a property binding and separate getter/setter bindings with the same names is just extra confuse.

Reply:
I prefer keeping it as one method with parameters.
Going with GetTotalWhatever is kinda tough cause what counts for total? Is it only damaging ones or should non-damaging ones count too? You'd probably have to put all the same arguments in, which makes it exactly the same just with a different name for no real reason.
That said, I'm totally okay with changing it so the default versions are just the object itself. I basically made the default versions replace the old GetTotalWhatever, plus I figured for GetWoundCount at least it's good to know how many wounds are on damaging attachables, cause you might wanna remove those, and I pretty much just made the rest match. But yeah, if it makes more sense for the no-argument version to just be the object, that's fine by me.

Regarding the weird numbers, yeah, I have no idea what the heck is going on there, there's definitely some maff problems. I'll look into it.

Regarding the property and definition for things, I don't actually like it either, but I kept it for a bit of backwards compatibility. Maybe it's better to kill it off and not have that weirdness, yeah.

Reply:
So regarding the weird numbers, everything seems fine now. You will find that gib wound limits on actors are 1.5x what they're defined as in INI cause of 4zK's script that sets gibwoundlimit to gibwoundlimit *1.5 for reasons (took me ages to remember about this haha). I think that script was probably interacting weirdly with GibWoundLimit not being what it was expected to be.
Also, I made a commit already that makes getting GibWoundLimit without arguments get the object's gib wound limit so everything seems good.

Only question left is, do we want a property that does the default as well as the definition that takes arguments, or just go with the definition and require another mod converter entry?